src/pygama/lgdo/table.py

```diff
@@ -205,3 +205,63 @@ def get_dataframe(self, cols: list[str] = None, copy: bool = False) -> pd.DataFr
                 df[col] = self[col].nda
 
         return df
+
+    def eval(self, expr_config: dict) -> Table:
+        """Apply column operations to the table and return a new table holding
+        the resulting columns.
+
+        Currently defers all the job to :meth:`pandas.DataFrame.eval`. This
+        might change in the future.
+
+        Parameters
+        ----------
+        expr_config
+            dictionary that configures expressions according the following
+            specification:
+
+            .. code-block:: js
+
+                {
+                    "O1": {
+                        "expression": "@p1 + @p2 * a**2",
+                        "parameters": {
+                            "p1": "2",
+                            "p2": "3"
+                        }
+                    },
+                    "O2": {
+                        "expression": "O1 - b"
+                    }
+                    // ...
+                }
+
+            where:
+
+            - ``expression`` is an expression string supported by
+              :meth:`pandas.DataFrame.eval` (see also `here
+              <https://pandas.pydata.org/pandas-docs/stable/user_guide/enhancingperf.html#expression-evaluation-via-eval>`_
+              for documentation).
+            - ``parameters`` is a dictionary of function parameters. Passed to
+              :meth:`pandas.DataFrame.eval` as `local_dict` argument.
+
+
+        Warning
+        -------
+        Blocks in `expr_config` must be ordered according to mutual dependency.
+        """
+        # don't want to modify input
+        df = self.get_dataframe().copy()
+        out_tbl = Table(size=self.size)
+
+        for out_var, spec in expr_config.items():
+            df.eval(
+                f"{out_var} = {spec['expression']}",
+                parser="pandas",
+                engine="numexpr",
+                local_dict=spec["parameters"] if "parameters" in spec else None,
+                inplace=True,
+            )
+
+            out_tbl.add_column(out_var, Array(df[out_var].to_numpy()))
```

Comment:
I was wondering why you didn't just do e.g. 
`out_var = df.eval(...).to_numpy()`
inside the `for` loop but I guess adding the column to the df allows you to have expressions that build on previous expressions (like in the test code). I think it would be good to add a comment in the code that it is structured this way for this purpose.

Reply:
Yes, that behavior is intended. I will add a comment.